### DISCUSSION TITLE
Schema generation at runtime

### DISCUSSION THREAD

**Discussion by MidasLamb** [async-graphql/async-graphql#930] [Q&A] [answered]
I'm trying to dynamically generate schemas at runtime. 

I was trying to approach this by implementing what the derive macro does manually. However I'm currently stuck because when trying to implement my QueryRoot, the `create_type_info` of `OutputType` does not contain any reference to `self`, which contains the information needed to determine which fields to add.

Is there any way to build the schemas at runtime, dynamically? Such that, based on the particular request, there is another schema.

**Comment by sunli829** [comment, marked as answer]
This is not possible, `Async-graphql` cannot support dynamic schema, I suggest you don't waste your time. 😂

**Reply by MidasLamb** [reply]
I thought as much, but wanted to check anyway, thanks!
